apps/testingaccessibility/src/components/app/navigation.tsx

```diff
@@ -5,37 +5,63 @@ import {isSellingLive} from 'utils/is-selling-live'
 import {useSession, signOut} from 'next-auth/react'
 import {Menu, Transition} from '@headlessui/react'
 import {NextRouter, useRouter} from 'next/router'
-import {Logo} from 'components/images'
 import isEmpty from 'lodash/isEmpty'
 import toast from 'react-hot-toast'
 import Link from 'next/link'
 import cx from 'classnames'
 import Image from 'next/image'
+import {useUser} from '../../context/user-context'
+import {getAbilityFromToken} from '../../server/ability'
+import {useQuery} from 'react-query'
+
+// TODO: This is a temporary solution to get the ability from the token
+//       It's named poorly and is clumping stuff together, but it works!
+function useRefinedSession() {
+  const {data, status} = useSession()
+  const {user} = useUser()
+
+  const isSignedIn = Boolean(data?.user)
+  const isLoadingUser = status === 'loading'
+
+  // this is all client side so it isn't secure secure but for display
+  // purposes it's fine
+  const {data: ability} = useQuery(
+    ['ability', user],
+    async () => await getAbilityFromToken(user),
+  )
```

Comment:
this works pretty well! we can use this all over.

Reply:
nice!